src/emulator.py

```diff
@@ -228,37 +228,90 @@ def read_and_confirm_mnemonic() -> None:
     client.open()
     time.sleep(SLEEP)
 
+    # Clicking continue button
     client.press_yes()
     time.sleep(SLEEP)
 
+    # Scrolling through all the 12 words on next three pages
+    for _ in range(3):
+        client.swipe_up()
+        time.sleep(SLEEP)
+
+    # Confirming that I have written the seed down
+    client.press_yes()
+    time.sleep(SLEEP)
+
+    # Retrieving the seed words for next "quiz"
     mnem = client.state().mnemonic_secret.decode("utf-8")
     mnemonic = mnem.split()
     time.sleep(SLEEP)
 
-    client.swipe_up()
+    # Answering 3 questions asking for a specific word
+    for _ in range(3):
+        index = client.read_reset_word_pos()
+        client.input(mnemonic[index])
+        time.sleep(SLEEP)
+
+    client.press_yes()
+    client.press_yes()
+    client.close()
+
+
+def read_and_confirm_mnemonic_shamir():
+    client = DebugLink(get_device().find_debug())
+    client.open()
     time.sleep(SLEEP)
 
-    client.swipe_up()
+    # Click Continue to begin Shamir setup process
+    client.press_yes()
+
+    # Clicking the minus button four times to set only one share (it starts at 5)
```

Comment:
I do not know if we need to do it now (but maybe why not?) but the number of shares and threshold should be configurable. In another words we should have it as arguments to the `emulator-read-and-confirm-mnemonic-shamir` command?

Reply:
Agreed, it could be really helpful for Suite tests. I will try to play with it quickly